File: src/main/java/io/github/kuroka3/playermanager/Commands/Mute.java
package io.github.kuroka3.playermanager.Commands;

import io.github.kuroka3.playermanager.Class.ManagedPlayer;
import io.github.kuroka3.playermanager.PlayerManager;
import io.github.kuroka3.playermanager.Utils.CaseManager;
import io.github.kuroka3.playermanager.Utils.JSONFile;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.OfflinePlayer;
import org.bukkit.block.CommandBlock;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.command.ConsoleCommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public class Mute implements CommandExecutor {

    private static String mod = ChatColor.GRAY + "[" + ChatColor.LIGHT_PURPLE + "!" + ChatColor.GRAY + "] " + ChatColor.RESET;

    @Override
    public boolean onCommand(@NotNull CommandSender sd, @NotNull Command cmd, @NotNull String label, @NotNull String[] args) {
        try {
            if (sd instanceof CommandBlock) {
                return true;
            }

            if (args.length == 0) {
                sd.sendMessage(mod + ChatColor.RED + "플레이어를 지정해 주십시오");
                return true;
            }

            if (!sd.hasPermission("playermanager.mute") && !(sd instanceof ConsoleCommandSender)) {
                sd.sendMessage(mod + ChatColor.RED + "이 명령어를 사용할 권한이 없습니다");
                return true;
            }

            JSONFile playerJson = new JSONFile(PlayerManager.getPlugin(PlayerManager.class).getDataFolder() + "/players.json");

            Player player2 = Bukkit.getServer().getPlayer(args[0]);

            ManagedPlayer target;

            if(player2 == null) {
                OfflinePlayer offlinePlayer = Bukkit.getServer().getOfflinePlayer(args[0]);
                if(offlinePlayer.hasPlayedBefore()) {
                    target = new ManagedPlayer(offlinePlayer, playerJson);
                } else {
                    sd.sendMessage(mod + ChatColor.RED + "해당 유저를 찾을 수 없습니다.");
                    return true;
                }
            } else {
                target = new ManagedPlayer(player2, playerJson);
            }

            if (target.isMute()) {
                sd.sendMessage(mod + ChatColor.RED + "해당 유저는 이미 뮤트 상태입니다");
                return true;
            }

            StringBuilder sb = new StringBuilder();

            for (int i = 1; i < args.length; i++) {
                sb.append(args[i]);
                if (i != args.length - 1) {
                    sb.append(" ");
                }
            }

            String reason = sb.toString();

            target.mute();

            String saveReason;

            if(reason.isEmpty()) {
                saveReason = "No Reason Given";
            } else {
                saveReason = reason;
            }

            String moder = null;

            if(sd instanceof Player player) {
                moder = player.getUniqueId().toString();
            } else if (sd instanceof ConsoleCommandSender) {
                moder = "Server";
            }

            if(!target.isOffline()) {
                sd.sendMessage(mod + ChatColor.YELLOW + target.getPlayer().getName() + ChatColor.RED + "님을 뮤트하였습니다: " + ChatColor.GOLD + saveReason);
                target.getPlayer().sendMessage(mod + ChatColor.RED + "관리자가 당신을 뮤트하였습니다: " + ChatColor.GOLD + saveReason);

                CaseManager.addCase(3, moder, target.getPlayer().getUniqueId(), saveReason, null);
            } else {
                sd.sendMessage(mod + ChatColor.YELLOW + target.getOfflinePlayer().getName() + ChatColor.RED + "님을 뮤트하였습니다: " + ChatColor.GOLD + saveReason);
                CaseManager.addCase(3, moder, target.getOfflinePlayer().getUniqueId(), saveReason, null);
            }
            return true;
        } catch (Exception e) {
            e.printStackTrace();
            sd.sendMessage(mod + ChatColor.RED + "Check Console: An Exception occured");
            return true;
        }
    }
}

File: src/main/kotlin/io/github/kuroka3/playermanagerkotlin/Commands/Mute.kt
package io.github.kuroka3.playermanagerkotlin.Commands

import io.github.kuroka3.playermanagerkotlin.Class.ManagedPlayer
import io.github.kuroka3.playermanagerkotlin.PlayerManagerKotlin
import io.github.kuroka3.playermanagerkotlin.Utils.JSONFile
import io.github.monun.kommand.KommandArgument
import io.github.monun.kommand.StringType
import io.github.monun.kommand.getValue
import io.github.monun.kommand.kommand
import net.kyori.adventure.text.Component.text
import net.kyori.adventure.text.TextComponent
import net.kyori.adventure.text.format.TextColor.color
import net.kyori.adventure.title.Title
import org.bukkit.Bukkit
import org.bukkit.entity.Player
import java.lang.Exception
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

object Mute {
    fun registerKommand() {

        val PlayerManagerKotlin = PlayerManagerKotlin.instance

        val mod: TextComponent = text("[").color(color(0x00aaaaaa)).append(
            text("!").color(color(0x00ff55ff))
        ).append(
            text("] ").color(color(0x00aaaaaa))
        )

        PlayerManagerKotlin.kommand {
            register("kotlinmute") {
                requires { (isPlayer && hasPermission("playermanager.mute")) || isConsole }

                val onlinePlayer = string().apply {
                    suggests {
                        Bukkit.getOnlinePlayers().forEach {
                            suggest(it.name)
                        }
                    }
                }

                val reasonString = string(StringType.GREEDY_PHRASE).apply {
                    suggests {
                        suggest(
                            listOf(
                                "No Reason Given",
                                "Reason"
                            )
                        )
                    }
                }

                then("target" to onlinePlayer) {
                    then("reason" to reasonString) {
                        executes {
                            try {
                                val target: String by it
                                val reason: String by it

                                val targetPlayer = Bukkit.getOfflinePlayer(target)

                                if (!targetPlayer.hasPlayedBefore()) throw IllegalArgumentException("Targeted player not found")

                                val managedPlayer: ManagedPlayer = ManagedPlayer(
                                    Bukkit.getOfflinePlayer(target),
                                    JSONFile("${PlayerManagerKotlin.dataFolder}/players.json")
                                )

                                managedPlayer.mute()

                                if (managedPlayer.p.isOnline) {
                                    val onp: Player = Bukkit.getPlayer(managedPlayer.p.uniqueId)!!
                                    onp.sendMessage(
                                        mod.append(
                                            text("관리자가 당신을 뮤트하였습니다: ").color(color(0x00ff5555))
                                        ).append(
                                            text(reason).color(color(0x00ffaa00))
                                        )
                                    )
                                }

                                sender.sendMessage(
                                    mod.append(
                                        text("${managedPlayer.p.name}").color(color(0x00ffff55))
                                    ).append(
                                        text("님을 뮤트하였습니다: ").color(color(0x00ff5555))
                                    ).append(
                                        text(reason).color(color(0x00ffaa00))
                                    )
                                )
                            } catch (e: Exception) {
                                when (e) {
                                    is IllegalArgumentException, is IllegalAccessException -> sender.sendMessage(
                                        mod.append(
                                            text("${e.message}").color(color(0x00ff55555))
                                        )
                                    )

                                    else -> {
                                        e.printStackTrace()
                                        sender.sendMessage(
                                            mod.append(
                                                text("Check Console: An Exception occured")
                                                    .color(color(0x00ff55555))
                                            )
                                        )
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}